spyder/plugins/editor/panels/scrollflag.py

```diff
@@ -82,6 +90,37 @@ def update_flag_colors(self, color_dict):
             self._facecolors[name] = QColor(color)
             self._edgecolors[name] = self._facecolors[name].darker(120)
 
+    def delayed_update_flags(self):
+        """Call update flags later."""
+        if self._update_list_timer:
+            return
+
+        self._update_list_timer = QTimer(self)
+        self._update_list_timer.setSingleShot(True)
+        self._update_list_timer.timeout.connect(self.update_flags)
+        self._update_list_timer.start(REFRESH_RATE)
+
+    def update_flags(self):
+        """Update flags list."""
+        self._update_list_timer = None
```

Comment:
Why to we make the timer none and then construct it. Could we not just create it once and use single shots and call start() every delayed update flags call?

Or even better not have that call at all?

Reply:
> Why to we make the timer none and then construct it. Could we not just create it once and use single shots and call start() every delayed update flags call?

Can I reuse a timer?

> Or even better not have that call at all?

I am not sure I understand what you suggest. The flags needs to be updated at some point surely?

Reply:
>Can I reuse a timer?

Sure, you start it as many times as you want.